From e6b9ea1743a9343117e090f1af711680c3498539 Mon Sep 17 00:00:00 2001
From: Rikkert Frederix <frederix@physik.uzh.ch>
Date: Tue, 28 Jun 2016 12:26:43 +0200
Subject: [PATCH 7/8] Fix in PDF reweighting (NLO process generation) if a PDF set has no
 error members, while reweight_pdf was set to True for that set.

---
 Template/NLO/Source/setrun.f                     |    4 ++++
 Template/NLO/SubProcesses/madfks_plot.f          |    4 ++++
 Template/NLO/SubProcesses/reweight_xsec_events.f |    4 ++++
 3 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/Template/NLO/Source/setrun.f b/Template/NLO/Source/setrun.f
index 77bc48b..239150a 100644
--- a/Template/NLO/Source/setrun.f
+++ b/Template/NLO/Source/setrun.f
@@ -204,6 +204,10 @@ C       Fill common block for Les Houches init info
 c fill the nmemPDF(i) array with the number of PDF error set. This we
 c get from LHAPDF.
          call numberPDFm(1,nmemPDF(1))
+         if (nmemPDF(1).eq.1) then
+            nmemPDF(1)=0
+            lpdfvar(1)=0
+         endif
       else
          nmemPDF(1)=0
       endif
diff --git a/Template/NLO/SubProcesses/madfks_plot.f b/Template/NLO/SubProcesses/madfks_plot.f
index 4a4bb5b..c5c8b7c 100644
--- a/Template/NLO/SubProcesses/madfks_plot.f
+++ b/Template/NLO/SubProcesses/madfks_plot.f
@@ -66,6 +66,10 @@ c     to "setrun")
                call initpdfsetbynamem(nn,lhaPDFsetname(nn))
                if (lpdfvar(nn)) then
                   call numberPDFm(nn,nmemPDF(nn))
+                  if (nmemPDF(nn).eq.1) then
+                     nmemPDF(nn)=0
+                     lpdfvar(nn)=.false.
+                  endif
                else
                   nmemPDF(nn)=0
                endif
diff --git a/Template/NLO/SubProcesses/reweight_xsec_events.f b/Template/NLO/SubProcesses/reweight_xsec_events.f
index 512c6a1..54fd9f6 100644
--- a/Template/NLO/SubProcesses/reweight_xsec_events.f
+++ b/Template/NLO/SubProcesses/reweight_xsec_events.f
@@ -87,6 +87,10 @@ c to "setrun")
                call initpdfsetbynamem(nn,lhaPDFsetname(nn))
                if (lpdfvar(nn)) then
                   call numberPDFm(nn,nmemPDF(nn))
+                  if (nmemPDF(nn).eq.1) then
+                     nmemPDF(nn)=0
+                     lpdfvar(nn)=.false.
+                  endif
                else
                   nmemPDF(nn)=0
                endif
-- 
1.7.1

